source/jormungandr/jormungandr/scenarios/journey_filter.py

```diff
@@ -200,84 +214,82 @@ def exceed_min_duration(current_section, journey, request,  min_duration):
            and current_section.duration < min_duration
 
 
-def _filter_too_short_heavy_journeys(journeys, request):
+def filter_too_short_heavy_journeys(journey, request):
     """
     we filter the journeys with use an "heavy" fallback mode if it's use only for a few minutes
     Heavy fallback mode are Bike and Car, bss is not considered as one.
     Typically you don't take your car for only 2 minutes
     """
     logger = logging.getLogger(__name__)
-    for journey in journeys:
-        if to_be_deleted(journey):
+
+    on_bss = False
+    is_debug = request.get('debug', False)
+    for s in journey.sections:
```

Comment:
Since we now have the duration for each mode, we can do an early return.
```
bike_duration==0: true
car_duration==0: true
car_duration<_min_car: false
bike_duration<_min_bike: false (only if bike and bss are distrinct in Durations)
```
we will only have to scan the whole journey when there is more car/bike than the min to check if one section isn't bellow this minimum

Reply:
LOL I understand your point and I tried really hard to optimize this, my final codes don't appear more compact than the original one :rofl: we can discuss about this later